How to pass  the switch --skip-kotlin-format with jhipster 7.0.1

## Overview of the issue
khipster crashed with --skip-kotlin-format switch

## Motivation for or Use Case

## Reproduce the error
```
❯ khipster --skip-kotlin-format
INFO! Using JHipster version installed globally
INFO! No custom sharedOptions found within blueprint: generator-jhipster-kotlin at /.../node_modules/generator-jhipster-kotlin
INFO! No custom commands found within blueprint: generator-jhipster-kotlin at /.../node_modules/generator-jhipster-kotlin
error: unknown option '--skip-kotlin-format'

❯ khipster --version
INFO! Using JHipster version installed globally
INFO! No custom sharedOptions found within blueprint: generator-jhipster-kotlin at /.../node_modules/generator-jhipster-kotlin
INFO! No custom commands found within blueprint: generator-jhipster-kotlin at /.../node_modules/generator-jhipster-kotlin
7.0.1

```
## Related issues

## Link to PR (if any)

## KHipster Version(s): 1.12.0

-   [] Checking this box is mandatory (this is just to show you read everything)

I tried the latest release for the first time today. I tried using the following to generate a microservices architecture.

```
khipster jdl reactive-ms
```

I did see the following during the generation process.

```
> Task :ktlintMainSourceSetFormat
/Users/mraible/kotlin-reactive/blog/src/main/kotlin/com/okta/developer/blog/repository/AuthorityRepository.kt:11:9: Not a valid Kotlin file (11:9 expecting member declaration) (cannot be auto-corrected)
"plain" report written to /Users/mraible/kotlin-reactive/blog/build/reports/ktlint/ktlintMainSourceSetFormat/ktlintMainSourceSetFormat.txt

> Task :ktlintMainSourceSetFormat FAILED
> Task :ktlintTestSourceSetFormatte-fns: sill extract date-fns@^1.27.2 extracted to /Users/mraible/kotlin-reactive/gate

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ktlintMainSourceSetFormat'.
> A failure occurred while executing org.jlleitschuh.gradle.ktlint.KtLintWorkAction
   > Process 'command '/Users/mraible/.sdkman/candidates/java/11.0.8.hs-adpt/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8.3/userguide/command_line_interface.html#sec:command_line_warnings
2 actionable tasks: 2 executed

BUILD FAILED in 45s
WARNING! Something went wrong while running ktlint formatter...ipt@4.2.3 extracted to /Users/mraible/kotlin-reactive/ga
WARNING! The generated application could not be committed to Git, as a Git repository could not be initialized.lin-reac
```

I do have JHipster 7.0.1 installed.

If I cd into the `gateway` directory and run `./gradlew`, it starts. The `blog` app does not.

```
> Task :ktlintMainSourceSetFormat FAILED
/Users/mraible/kotlin-reactive/blog/src/main/kotlin/com/okta/developer/blog/repository/AuthorityRepository.kt:11:9: Not a valid Kotlin file (11:9 expecting member declaration) (cannot be auto-corrected)
"plain" report written to /Users/mraible/kotlin-reactive/blog/build/reports/ktlint/ktlintMainSourceSetFormat/ktlintMainSourceSetFormat.txt
```

The code in this file is:

```kotlin
package com.okta.developer.blog.repository

import com.okta.developer.blog.domain.Authority
import org.springframework.data.neo4j.repository.ReactiveNeo4jRepository
import reactor.core.publisher.Flux
/**
* Spring Data Neo4j repository for the {@link Authority} entity.
*/

interface AuthorityRepository : ReactiveNeo4jRepository<Authority, String> {
        <Authority> findAll(): Flux
}
```

For `store`, it fails to start too:

```
> Task :compileKotlin FAILED
e: /Users/mraible/kotlin-reactive/store/src/main/kotlin/com/okta/developer/store/config/SecurityConfiguration.kt: (115, 25): Unresolved reference: extractAuthorityFromClaims
```

--- a/generators/app/index.js
+++ b/generators/app/index.js
@@ -23,6 +23,13 @@ const { displayLogo } = require('./util');
 module.exports = class extends BaseGenerator {
     constructor(args, opts) {
         super(args, { fromBlueprint: true, ...opts });
+
+        // This adds support for a `--skip-ktlint-format` flag
+        this.option('skip-ktlint-format', {
+            desc: 'Indicates to skip formatting using ktlint',
+            type: Boolean,
+            defaults: false,
+        });
     }
 
     get initializing() {

--- a/generators/entity-server/templates/src/main/kotlin/package/domain/Entity.kt.ejs
+++ b/generators/entity-server/templates/src/main/kotlin/package/domain/Entity.kt.ejs
@@ -428,8 +428,6 @@ for (const field of filteredFields) {
     <%_ } _%>
     <%_ if (databaseType === 'sql' && reactive) { _%>
     @Transient
-    <%_ } else if (databaseType === 'neo4j') { _%>
-    @Relationship("<%= relationshipName %>")
     <%_ } _%>
     var <%= relationshipFieldNamePlural %>: MutableSet<<%= asEntity(otherEntityNameCapitalized) %>>? = mutableSetOf(),
 

--- a/generators/server/index.js
+++ b/generators/server/index.js
@@ -32,12 +32,6 @@ module.exports = class extends ServerGenerator {
         if (!jhContext) {
             this.error("This is a JHipster blueprint and should be used only like 'jhipster --blueprints myblueprint')}");
         }
-        // This adds support for a `--skip-ktlint-format` flag
-        this.option('skip-ktlint-format', {
-            desc: 'Indicates to skip formatting using ktlint',
-            type: Boolean,
-            defaults: false,
-        });
     }
 
     get initializing() {
@@ -80,7 +74,7 @@ module.exports = class extends ServerGenerator {
         const phaseFromJHipster = super._install();
         const myCustomPhaseSteps = {
             lintFiles() {
-                if (this.options['skip-ktlint-format']) {
+                if (this.options.skipKtlintFormat) {
                     this.log('Skipping ktlint format...');
                     return;
                 }

--- a/generators/server/templates/src/main/kotlin/package/config/SecurityConfiguration_reactive.kt.ejs
+++ b/generators/server/templates/src/main/kotlin/package/config/SecurityConfiguration_reactive.kt.ejs
@@ -23,12 +23,10 @@ import <%= packageName %>.security.ADMIN
 import <%= packageName %>.security.jwt.JWTFilter
 import <%= packageName %>.security.jwt.TokenProvider
 <%_ } _%>
-<%_ if (authenticationType === 'oauth2' && applicationType !== 'microservice') { _%>
-import <%= packageName %>.security.extractAuthorityFromClaims
-<%_ } _%>
 <%_ if (authenticationType === 'session' || authenticationType === 'oauth2') { _%>
     <%_ if (authenticationType === 'oauth2') { _%>
 import <%= packageName %>.security.SecurityUtils
+import <%= packageName %>.security.extractAuthorityFromClaims
 import <%= packageName %>.security.oauth2.AudienceValidator
 import <%= packageName %>.security.oauth2.JwtGrantedAuthorityConverter
 import org.springframework.security.authentication.AbstractAuthenticationToken

--- a/generators/server/templates/src/main/kotlin/package/domain/User.kt.ejs
+++ b/generators/server/templates/src/main/kotlin/package/domain/User.kt.ejs
@@ -376,7 +376,7 @@ class <%= asEntity('User') %> (
     <%_ if (enableHibernateCache) { _%>
     @Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
     <%_ } _%>
-    var persistentTokens: MutableSet<PersistentToken> = mutableSetOf()<% } %><% if (databaseType === 'sql' || databaseType === 'mongodb' || databaseType === 'couchbase') { %>,
+    var persistentTokens: MutableSet<PersistentToken> = mutableSetOf()<% } %><% if (databaseType === 'sql' || databaseType === 'mongodb' || databaseType === 'neo4j' || databaseType === 'couchbase') { %>,
     createdBy: String? = null,
     createdDate: Instant? = Instant.now(),
     lastModifiedBy: String? = null,

--- a/generators/server/templates/src/main/kotlin/package/repository/AuthorityRepository.kt.ejs
+++ b/generators/server/templates/src/main/kotlin/package/repository/AuthorityRepository.kt.ejs
@@ -54,6 +54,6 @@ import org.springframework.data.neo4j.repository.<% if (reactive) { %>Reactive<%
 interface AuthorityRepository : <% if (databaseType === 'sql') { %><% if (reactive) { %>R2dbc<% } else { %>Jpa<% } %>Repository<% } else { %><% if (reactive) { %>Reactive<% } %><% if (databaseType === 'mongodb') { %>MongoRepository<% } %><% if (databaseType === 'neo4j') { %>Neo4jRepository<% } %><% if (databaseType === 'couchbase') { %>N1qlCouchbaseRepository<% } %><% } %><Authority, String> {
     <%_ if (databaseType === 'neo4j') { _%>
     <% if (!reactive) { %>// See https://github.com/neo4j/sdn-rx/issues/51<%_ } _%>
-    <<%= asEntity('Authority') %>> findAll(): <% if (reactive) { %>Flux<% } else { %>List<% } %>
+    override fun findAll(): <% if (reactive) { %>Flux<% } else { %>List<% } %><<%= asEntity('Authority') %>>
     <%_ } _%>
 }

--- a/generators/server/templates/src/main/kotlin/package/repository/UserRepository.kt.ejs
+++ b/generators/server/templates/src/main/kotlin/package/repository/UserRepository.kt.ejs
@@ -195,7 +195,7 @@ interface UserRepository : <% if (databaseType === 'sql') { %>JpaRepository<<%= 
 
     <%_ if (databaseType === 'neo4j') { _%>
     <% if (!reactive) { %>// See https://github.com/neo4j/sdn-rx/issues/51<% } %>
-    fun findAll(): <%= listOrFlux %><<%= asEntity('User') %>>
+    override fun findAll(): <%= listOrFlux %><<%= asEntity('User') %>>
     <%_ } _%>
     <%_ if (databaseType === 'sql') { _%>
 

--- a/generators/server/templates/src/main/kotlin/package/service/UserService.kt.ejs
+++ b/generators/server/templates/src/main/kotlin/package/service/UserService.kt.ejs
@@ -641,7 +641,6 @@ class UserService<% if (databaseType !== 'no') { %>(
                     <%_ } else { _%>
                     userRepository.save(user)
                     <%_ } _%>
-                    userRepository.save(user)
                         .flatMap { user ->
                             Flux.fromIterable(user.authorities)
                                 .flatMap { user.id?.let { it1 -> it.name?.let { it2 -> userRepository.saveUserAuthority(it1, it2) } } }

